packages/fscomponents/src/components/ModalHalfScreen.tsx

```diff
@@ -21,33 +23,96 @@ const styles = StyleSheet.create({
   }
 });
 
+interface ModalHalfScreenAnimationConfig {
+  contentOffsetY: {
+    inputRange: number[];
+    outputRange: number[];
+  };
+
+  heightFromProps: {
+    height: number;
+    top?: number;
+    bottom?: number;
+  } | {};
+}
+
+export enum ModalHalfScreenPosition {
+  Top = 'top',
+  Center = 'center',
+  Bottom = 'bottom'
+}
+
 export interface ModalHalfScreenProps {
   visible?: boolean;
   height?: number;
   backgroundAccessibilityTraits?: AccessibilityTraits;
   backgroundAccessibilityComponentType?: AccessibilityComponentType;
   backgroundAccessibilityLabel?: string;
   onRequestClose: () => void;
+  position?: (
+    ModalHalfScreenPosition.Top |
+    ModalHalfScreenPosition.Center |
+    ModalHalfScreenPosition.Bottom
+  );
 }
 
 export interface ModalHalfScreenState {
   visible: boolean;
   contentOffset: Animated.Value;
+  height: number;
+  statusBarHeight: number;
 }
 
 export class ModalHalfScreen extends PureComponent<ModalHalfScreenProps, ModalHalfScreenState> {
   constructor(props: ModalHalfScreenProps) {
     super(props);
     this.state = {
       visible: false,
-      contentOffset: new Animated.Value(0)
+      contentOffset: new Animated.Value(0),
+      height: 0,
```

Comment:
You should be able to set this to the height prop. If that is being set incorrectly, it's a problem with the parent component.